compiler/rustc_typeck/src/check/generator_interior/drop_ranges/cfg_build.rs

```diff
@@ -334,7 +373,8 @@ impl<'a, 'tcx> Visitor<'tcx> for DropRangeVisitor<'a, 'tcx> {
             }
             ExprKind::Break(hir::Destination { target_id: Ok(target), .. }, ..)
             | ExprKind::Continue(hir::Destination { target_id: Ok(target), .. }, ..) => {
-                self.drop_ranges.add_control_edge_hir_id(self.expr_index, target);
+                self.drop_ranges
+                    .add_control_edge_hir_id(self.expr_index, self.find_target_expression(target));
```

Comment:
oh, I guess it is used for both break/continue-- that confuses me a bit. It would be good to give an example

```
'a: {
   ...
   break 'a
}

'a: loop {
   ...
   continue 'a
}
```

Reply:
It's probably confusing because it's wrong. Here's an example that fails:
```rust
let mut arr = [Ptr];
let mut count = 0;
drop(arr);
while count < 3 {
    count += 1;
    yield;
    if b {
        arr = [Ptr];
        continue;
    }
}
```
I mistakenly assumed that `Break` and `Continue` worked like `goto` where the `Destination` field pointed at the expression where control flow transferred to. It actually points to the block being broken or continued. It mostly worked because of the way post-order numbering worked.

Reply:
I just uploaded a new commit that handles continue correctly.